povm_toolbox/quantum_info/base_dual.py

```diff
@@ -0,0 +1,67 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""TODO."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TypeVar
+
+import numpy as np
+from qiskit.quantum_info import Operator, SparsePauliOp
+
+from .base_frame import BaseFrame
+
+# type of the labels used to specify dual operators
+LabelT = TypeVar("LabelT")
+
+
+class BaseDUAL(BaseFrame[LabelT], ABC):
+    """Abstract base class that contains all methods that any specific DUAL subclass should implement."""
+
+    @property
+    def n_outcomes(self) -> int:
+        """Give the number of outcomes of the DUAL."""
+        return self.n_operators
+
+    @abstractmethod
+    def get_omegas(
+        self,
+        obs: SparsePauliOp | Operator,
+        outcome_idx: LabelT | set[LabelT] | None = None,
+    ) -> float | dict[LabelT, float] | np.ndarray:
+        """Return the decomposition weights of observable `obs` into the POVM effects to which `self` is a dual."""
```

Comment:
```suggestion
        """Return the decomposition weights of the provided observable into the POVM effects to which ``self`` is a dual."""
```
The docstrings are RST and not Markdown. Please correct this also in the other docstrings further down in this file (and possibly other new ones, too).